### PR TITLE
Core 17587

### PR DESCRIPTION
**Pull request by zefklop**
## Purpose

Fix stuff.
JIRA issue: [CORE-17587](https://jira.reactos.org/browse/CORE-17587)

## Proposed changes

- Fix MI_MAKE_HARDWARE_PTE_KERNEL: Currently it creates read/write PTEs whatever the protection it is handed to.
- Try fix CORE-17587 with debug prints to see what happens on test bot

## TODO

- [x] Remove debug prints

### CODE REVIEW

File: ntoskrnl/mm/i386/page.c [hunk 1]
```diff
@@ -710,11 +785,29 @@ MmSetPageProtect(PEPROCESS Process, PVOID Address, ULONG flProtect)
...
+    {
+        DPRINT1("TempPte.u.Long 0x%x, Old implementation 0x%x\n", TempPte.u.Long & 0xFFF, ProtectToPTE(flProtect));
+        ASSERT((TempPte.u.Long & 0xFFF) == ProtectToPTE(flProtect));
+    }
+
     /* Keep dirty & accessed bits */
     TempPte.u.Hard.Accessed = PointerPte->u.Hard.Accessed;
     TempPte.u.Hard.Dirty = PointerPte->u.Hard.Dirty;
 
+
```
Comment (SergeGautherie) [inline ntoskrnl/mm/i386/page.c:810]: Unwanted?

File: ntoskrnl/mm/i386/page.c [hunk 2]
```diff
@@ -710,11 +785,29 @@ MmSetPageProtect(PEPROCESS Process, PVOID Address, ULONG flProtect)
...
+    /* Sanity check */
+    ASSERT(PointerPte->u.Hard.Owner == 1);
+
+    TempPte.u.Long = 0;
+    TempPte.u.Hard.PageFrameNumber = PointerPte->u.Hard.PageFrameNumber;
+    TempPte.u.Long |= MmProtectToPteMask[ProtectionMask];
+    TempPte.u.Hard.Owner = 1;
+
+    /* Only set valid bit if we have to */
+    if ((ProtectionMask != MM_NOACCESS) && !FlagOn(ProtectionMask, MM_GUARDPAGE))
```
Comment (SergeGautherie) [inline ntoskrnl/mm/i386/page.c:797]: Use `FlagOn()` elsewhere too?

File: ntoskrnl/mm/i386/page.c [hunk 3]
```diff
@@ -19,6 +19,66 @@
...
+    }
+    if (flProtect & PAGE_NOCACHE)
+    {
+        Attributes = Attributes | PA_CD;
+    }
+    if (flProtect & PAGE_WRITETHROUGH)
+    {
+        Attributes = Attributes | PA_WT;
+    }
+    return(Attributes);
```
Comment (HBelusca) [inline ntoskrnl/mm/i386/page.c:79]:
```suggestion
    return Attributes;
```

File: ntoskrnl/mm/ARM3/miarm.h [hunk 4]
```diff
@@ -830,7 +845,10 @@ MI_MAKE_HARDWARE_PTE_USER(IN PMMPTE NewPte,
     /* Start fresh */
     NewPte->u.Long = 0;
 
-    /* Set the protection and page */
```
[the remark this answers is not in the document]
Comment (zefklop) [reply ntoskrnl/mm/ARM3/miarm.h:833]: Variable names are explicit enough.

**Pull request merged**